src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java

```diff
@@ -72,6 +111,109 @@ BEFORE_LABEL_ACAS, createDocLinkBinary(d), acasCount.incrementAndReturnValue()))
+            for (VettingJurCodesTypeItem codeItem : caseData.getVettingJurisdictionCodeCollection()) {
+                JurCodesType newCode = new JurCodesType();
+                newCode.setJuridictionCodesList(codeItem.getValue().getEt1VettingJurCodeList());
+                JurCodesTypeItem codesTypeItem = new JurCodesTypeItem();
+                codesTypeItem.setValue(newCode);
+                codesTypeItem.setId(UUID.randomUUID().toString());
+                caseData.getJurCodesCollection().add(codesTypeItem);
+            }
+        }
+
+        if (caseData.getJurCodesCollection().stream()
+            .anyMatch(c -> JUR_CODE_CONCILIATION_TRACK_OP.contains(c.getValue().getJuridictionCodesList()))) {
+            return String.format(TRACk_ALLOCATION_HTML, TRACK_OPEN);
+        } else if (caseData.getJurCodesCollection().stream()
+            .anyMatch(c -> JUR_CODE_CONCILIATION_TRACK_ST.contains(c.getValue().getJuridictionCodesList()))) {
+            return String.format(TRACk_ALLOCATION_HTML, TRACK_STANDARD);
+        } else if (caseData.getJurCodesCollection().stream()
+            .anyMatch(c -> JUR_CODE_CONCILIATION_TRACK_SH.contains(c.getValue().getJuridictionCodesList()))) {
+            return String.format(TRACk_ALLOCATION_HTML, TRACK_SHORT);
+        } else {
+            return String.format(TRACk_ALLOCATION_HTML, TRACK_NO);
+        }
+    }
+
+    public void populateTribunalOfficeFields(CaseData caseData) {
+        String managingOffice = caseData.getManagingOffice();
+        String tribunalLocation = TribunalOffice.isScotlandOffice(managingOffice)
+            ? TRIBUNAL_Scotland : TRIBUNAL_ENGLAND;
+        caseData.setTribunalAndOfficeLocation(String
+            .format(TRIBUNAL_OFFICE_LOCATION, tribunalLocation, managingOffice));
+        caseData.setRegionalOffice(String.format(TRIBUNAL_LOCATION_LABEL, tribunalLocation));
+        caseData.setRegionalOfficeList(populateRegionalOfficeList(tribunalLocation, managingOffice));
+    }
+
+    private DynamicFixedListType populateRegionalOfficeList(String tribunalLocation, String managingOffice) {
+        if (tribunalLocation.equals(TRIBUNAL_ENGLAND)) {
+            return DynamicFixedListType.from(TribunalOffice.ENGLANDWALES_OFFICES.stream()
+                .filter(tribunalOffice -> !tribunalOffice.getOfficeName().equals(managingOffice))
+                .map(tribunalOffice ->
+                    DynamicValueType.create(tribunalOffice.getOfficeName(), tribunalOffice.getOfficeName()))
+                .collect(Collectors.toList()));
+        } else {
+            return DynamicFixedListType.from(TribunalOffice.SCOTLAND_OFFICES.stream()
+                .filter(tribunalOffice -> !tribunalOffice.getOfficeName().equals(managingOffice))
+                .map(tribunalOffice ->
+                    DynamicValueType.create(tribunalOffice.getOfficeName(), tribunalOffice.getOfficeName()))
+                .collect(Collectors.toList()));
+        }
```

Comment:
This is duplicating code.  Better to find a way to avoid duplication, otherwise things could go badly wrong later if one is changed but the dev forgets to change the other.  Happy to look at this with you if you'd like?